### ISSUE TITLE
UnauthorizedAccessException D:\Windows\system32\Buffer.bookmark on Azure App Service (IIS)

### ISSUE THREAD

**Issue by lukasz-pyrzyk** [FantasticFiasco/serilog-sinks-http#267]
**Describe the bug**

The durable feature doesn't work within Azure App Service environment when app is served via IIS. I tried to enable a durable logger on an app running on .NET Framework 4.8 & IIS and unfortunately, it silently fails with the internal Serilog message:

> 2022-07-22T07:19:13  PID[5944] Error       Serilog internal error: 2022-07-22T07:19:13.8901164Z Exception while emitting periodic batch from Serilog.Sinks.Http.Private.Durable.HttpLogShipper: System.UnauthorizedAccessException: Access to the path 'D:\Windows\system32\Buffer.bookmark' is denied.at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)at Serilog.Sinks.Http.Private.Durable.BookmarkFile..ctor(String bookmarkFileName)at Serilog.Sinks.Http.Private.Durable.HttpLogShipper.<OnTick>d__14.MoveNext()

**To Reproduce**

Deploy any application with a durable logger to the Azure App Service and run it on in-process IIS. 

A side note, logger works well on Azure App Services when .NET 6 where the out-of-process hosting model is used.

**Comment by FantasticFiasco** [comment]
Sorry for the delay in response time, we're currently having summer vacations here in Sweden. I'll take a look at your issue when the vacations are over, but have you tried configuring the sink regarding where to save the buffer files? I haven't tried running the sink in Azure App Services but would expect that the path `D:\Windows\system32` requires Admin access, which I assume the app doesn't have.

A quick search gave me [Operating system functionality on Azure App Service](https://docs.microsoft.com/en-us/azure/app-service/operating-system-functionality#file-access) describing how you as an app developer can write files to disk. Does this documentation help you?

**Comment by lukasz-pyrzyk** [comment]
HI @FantasticFiasco, sorry for the late reply, also holidays :) Thank you for your answer. 

I've solved my issue by extending our library wrapper with a mechanism that detects running on Azure. In such case, a base path for logs is changed accordingly to the `%HOME%` + path for the app directory. 

I think we can solve this ticket. I've created it because at first, I missed that the log directory can be changed with `BufferBaseFileName`. Property name mentions `file name`, so I didn't think it takes directory into consideration. However, it is well described in the property description, so I'm sorry.
